src/amoc_lager_logger_formatter.erl

```diff
@@ -0,0 +1,129 @@
+                    {pid, Pid} = lists:keyfind(pid, 1, Started),
+                    {"Supervisor ~w started as ~p at pid ~w", [Name, error_logger_lager_h:format_mfa({Module, Function, Args}), Pid]};
+                {id, ChildID} ->
+                    case lists:keyfind(pid, 1, Started) of
+                        {pid, Pid} ->
+                            {"Supervisor ~w started child ~p at pid ~w", [Name, ChildID, Pid]};
+                        false ->
+                            %% children is a list of pids for some reason? and we only get the count
+                            {nb_children, ChildCount} = lists:keyfind(nb_children, 1, Started),
+                            {"Supervisor ~w started ~b children ~p", [Name, ChildCount, ChildID]}
+                    end
+            end
+    end;
+report_cb(#{label := {supervisor, _Error}, report := Report}) ->
+    {supervisor, Name} = lists:keyfind(supervisor, 1, Report),
+    {reason, Reason} = lists:keyfind(reason, 1, Report),
+    Formatted = error_logger_lager_h:format_reason(Reason),
+    {errorContext, ErrorContext} = lists:keyfind(errorContext, 1, Report),
+    {offender, Offender} = lists:keyfind(offender, 1, Report),
+    case lists:keyfind(mod, 1, Offender) of
+        {mod, _Mod} ->
+            {pid, Pid} = lists:keyfind(pid, 1, Offender),
+            %% this comes from supervisor_bridge
+            {"Supervisor ~w had ~p ~p with reason ~s", [Name, Pid, ErrorContext, Formatted]};
+        false ->
+            {id, ChildID} = lists:keyfind(id, 1, Offender),
+            case lists:keyfind(pid, 1, Offender) of
+                {pid, Pid} ->
+                    {"Supervisor ~w had ~p ~p ~p with reason ~s", [Name, ChildID, Pid, ErrorContext, Formatted]};
+                false ->
+                    {"Supervisor ~w had ~p ~p with reason ~s", [Name, ChildID, ErrorContext, Formatted]}
+            end
+    end;
+report_cb(#{label := {application_controller, progress}, report := Report}) ->
+    case application:get_env(lager, suppress_application_start_stop, false) of
+        true -> {"", []};
+        false ->
+            {application, Name} = lists:keyfind(application, 1, Report),
+            {started_at, Node} = lists:keyfind(started_at, 1, Report),
+            {"Application ~w started on node ~w", [Name, Node]}
+    end;
+report_cb(#{label := {application_controller, exit}, report := Report}) ->
+    {exited, Reason} = lists:keyfind(exited, 1, Report),
+    case application:get_env(lager, suppress_application_start_stop) of
+        {ok, true} when Reason == stopped ->
+            {"", []};
+        _ ->
+            {application, Name} = lists:keyfind(application, 1, Report),
+            Formatted = error_logger_lager_h:format_reason(Reason),
+            {"Application ~w exited with reason: ~s", [Name, Formatted]}
+    end;
+report_cb(Other) ->
+    {"Unexpected logger report_cb: ~w", [Other]}.
```

Comment:
As far as I know this is our addition to the formater, right? I think it makes unexpected logs unreadable in some case. For instance I have this error (it's my fault in Erlang and docsh configuration) printed by pure logger:

```
(amoc@MichalPiotrowski-MacBookPro2018)1> 09:25:05.869 [error] file:path_eval(["/Users/michalpiotrowski"],".erlang"): error on line 50: 50: evaluation failed with reason error:undef and stacktrace [{docsh,activated,[path],[]},{erl_eval,do_apply,6,[{file,[101,114,108,95,101,118,97,108,46,101,114,108]},{line,680}]},{erl_eval,exprs,5,[{file,[101,114,108,95,101,118,97,108,46,101,114,108]},{line,126}]},{erl_eval,expr,5,[{file,[101,114,108,95,101,118,97,108,46,101,114,108]},{line,462}]},{file,eval_stream2,6,[{file,[102,105,108,101,46,101,114,108]},{line,1393}]},{file,path_eval,3,[{file,[102,105,108,101,46,101,114,108]},{line,1059}]},{c,f_p_e,2,[{file,[99,46,101,114,108]},{line,693}]},{init,eval_script,2,[]}]
```

Which is readable and I know what to do about it. When the lager/logger integration is enabled the above log looks like the following:
```
2019-10-30 09:54:30.965 [error] <0.8.0> Unexpected logger report_cb: #{args => [[[47,85,115,101,114,115,47,109,105,99,104,97,108,112,105,111,116,114,111,119,115,107,105]],[46,101,114,108,97,110,103],50,[[53,48],58,32,101,118,97,108,117,97,116,105,111,110,32,102,97,105,108,101,100,32,119,105,116,104,32,114,101,97,115,111,110,32,[101,114,114,111,114],58,[117,110,100,101,102],32,97,110,100,32,115,116,97,99,107,116,114,97,99,101,32,[91,[[123,[[100,111,99,115,104],44,[97,99,116,105,118,97,116,101,100],44,[91,[[112,97,116,104]],93],44,[91,93]],125],44,[123,[[101,114,108,95,101,118,97,108],44,[100,111,95,97,112,112,108,121],44,[54],44,[91,[[123,[[102,105,108,101],44,[91,[[49,48,49],44,[49,49,52],44,[49,48,56],44,[57,53],44,[49,48,49],44,[49,49,56],44,[57,55],44,[49,48,56],44,[52,54],44,[49,48,49],44,[49,49,52],44,[49,48,56]],93]],125],44,[123,[[108,105,110,101],44,[54,56,48]],125]],93]],125],44,[123,[[101,114,108,95,101,118,97,108],44,[101,120,112,114,115],44,[53],44,[91,[[123,[[102,105,108,101],44,[91,[[49,48,49],44,[49,...],...],...]],...],...],...]],...],...],...]]],...}
```
This is very cryptic and it's close to impossible to figure out what's wrong.

Current master branch of Amoc produces the same output as my first example.

I'm not sure if trying to unify lager and logger logs makes sense, at least for Amoc.